src/source/avro.rs

```diff
@@ -55,6 +88,19 @@ impl AvroStreamHandler {
             Err(_) => return,
         };
 
+        let header_size_in_bytes = mem::size_of::<Header>();
+        if payload_size_in_bytes <= header_size_in_bytes {
+            warn!("Invalid payload detected!  Discarding!");
+            return;
+        }
+
+        let mut header_raw = Vec::with_capacity(header_size_in_bytes);
+        if reader.read_exact(&mut header_raw).is_err() {
+            warn!("Failed to read header from stream!");
+            return;
+        }
+        let header : Header = header_raw.into();
+
         let mut buf = Vec::with_capacity(payload_size_in_bytes);
```

Comment:
Since the intention of the length is to have it include the header bits, and we have already read that, we should be subtracting the header size in bytes from the payload size in bytes to arrive at the remaining size to read

Reply:
Fixed.